tests/test_cargo_test.rs

```diff
@@ -1206,6 +1206,46 @@ test!(test_no_harness {
                        .with_stdout(&format!("\
 {compiling} foo v0.0.1 ({dir})
 {running} target[..]bar-[..]
+
+running 0 tests
+
+test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured
+
+",
+                       compiling = COMPILING, running = RUNNING,
+                       dir = p.url())));
+});
+
+test!(test_no_harness_in_lib {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [package]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [lib]
+            name = "foo"
+            harness = false
+        "#)
+        .file("src/lib.rs", "#[cfg(test)] fn main() {}");
```

Comment:
Restoring the old functionality of the function you commented about still makes it pass with `#[cfg(foo)]`